ruma-events/src/room/redaction.rs

```diff
@@ -64,84 +67,11 @@ pub struct RedactionEventContent {
-impl ruma_events::MessageEventContent for RedactionEventContent {}
-
-#[cfg(test)]
-mod tests {
-    use std::{
-        convert::TryFrom,
-        time::{Duration, UNIX_EPOCH},
-    };
-
-    use matches::assert_matches;
-    use ruma_identifiers::{EventId, RoomId, UserId};
-    use serde_json::{from_value as from_json_value, json, to_value as to_json_value};
-
-    use super::{RedactionEvent, RedactionEventContent};
-    use crate::{EventJson, UnsignedData};
-
-    #[test]
-    fn serialization() {
-        let event = RedactionEvent {
-            content: RedactionEventContent { reason: Some("redacted because".into()) },
-            redacts: EventId::try_from("$h29iv0s8:example.com").unwrap(),
-            event_id: EventId::try_from("$h29iv0s8:example.com").unwrap(),
-            origin_server_ts: UNIX_EPOCH + Duration::from_millis(1),
-            room_id: RoomId::try_from("!roomid:room.com").unwrap(),
-            sender: UserId::try_from("@carl:example.com").unwrap(),
-            unsigned: UnsignedData::default(),
-        };
-
-        let json = json!({
-            "content": {
-                "reason": "redacted because"
-            },
-            "event_id": "$h29iv0s8:example.com",
-            "origin_server_ts": 1,
-            "redacts": "$h29iv0s8:example.com",
-            "room_id": "!roomid:room.com",
-            "sender": "@carl:example.com",
-            "type": "m.room.redaction",
-        });
-
-        assert_eq!(to_json_value(&event).unwrap(), json);
-    }
-
-    #[test]
-    fn deserialization() {
-        let e_id = EventId::try_from("$h29iv0s8:example.com").unwrap();
-
-        let json = json!({
-            "content": {
-                "reason": "redacted because"
-            },
-            "event_id": "$h29iv0s8:example.com",
-            "origin_server_ts": 1,
-            "redacts": "$h29iv0s8:example.com",
-            "room_id": "!roomid:room.com",
-            "sender": "@carl:example.com",
-            "type": "m.room.redaction",
-        });
-
-        assert_matches!(
-            from_json_value::<EventJson<RedactionEvent>>(json)
-                .unwrap()
-                .deserialize()
-                .unwrap(),
-            RedactionEvent {
-                content: RedactionEventContent {
-                    reason: Some(reason),
-                },
-                sender,
-                event_id, origin_server_ts, redacts, room_id, unsigned,
-            } if reason == "redacted because" && redacts == e_id
-                && event_id == e_id
-                && sender == "@carl:example.com"
-                && origin_server_ts == UNIX_EPOCH + Duration::from_millis(1)
-                && room_id == RoomId::try_from("!roomid:room.com").unwrap()
-                && unsigned.is_empty()
-        );
-    }
-}
+impl RoomEventContent for RedactionEventContent {}
+
+impl MessageEventContent for RedactionEventContent {}
+
+// A redacted custom event must satisfy both state and message events.
```

Comment:
We don't have this kind of comment for the `impl`s above. Also I'm not sure about the wording. If we have this at all, it should say something like `can be used as either … or …` or `can be used both as … and …`.